File: chain/java/FryingChef.java
package chain.java;

public class FryingChef implements Chef {
    private Chef nextChef;

    @Override
    public Chef setNextChef(Chef chef) {
        this.nextChef = chef;
        return chef;
    }

    @Override
    public void cook(String dish) {
        System.out.println("FryingChef is frying ingredients for: " + dish);
        if (nextChef != null) {
            nextChef.cook(dish);
        } else {
            System.out.println("No chef available to cook: " + dish);
        }
    }
}

File: chain/golang/frying_chef.go
package main

import "fmt"

type FryingChef struct {
	next Chef
}

func (f *FryingChef) Execute(d *Dish) {
	fmt.Printf("Frying %s...\n", d.Name)
	d.IsFried = true
	if f.next != nil {
		f.next.Execute(d)
	}
}

func (f *FryingChef) SetNext(next Chef) {
	f.next = next
}
